plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/selectionwizard/NewMenuWizard.java

```diff
@@ -11,37 +11,93 @@
+import org.eclipse.reddeer.swt.impl.button.FinishButton;
+import org.eclipse.reddeer.swt.impl.button.NoButton;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
 
 /**
- * Represents wizard which can be found in new wizard dialog (menu 'File -&gt; New -&gt; Other...').
+ * Represents wizard which can be found in new wizard dialog (menu 'File -&gt;
+ * New -&gt; Other...').
+ * 
  * @author rawagner
+ * @contributor jkopriva@redhat.com
  *
  */
-public abstract class NewMenuWizard extends AbstractSelectionWizardDialog{
-	
+public abstract class NewMenuWizard extends AbstractSelectionWizardDialog {
+
 	/**
 	 * Constructs new Wizard that can be found in NewWizard.
-	 * @param shellText new wizard text when next is clicked (ie 'New File')
+	 * 
+	 * @param shellText      new wizard text when next is clicked (ie 'New File')
 	 * @param wizardCategory new wizard category (ie 'General')
-	 * @param wizardName new wizard name (ie 'File')
+	 * @param wizardName     new wizard name (ie 'File')
 	 */
 	public NewMenuWizard(String shellText, String wizardCategory, String wizardName) {
 		super(shellText, wizardCategory, wizardName);
 	}
-	
+
 	/**
 	 * Constructs new Wizard that can be found in NewWizard.
-	 * @param shellText new wizard text when next is clicked (ie 'New File')
+	 * 
+	 * @param shellText  new wizard text when next is clicked (ie 'New File')
 	 * @param wizardPath wizard path in new wizard (ie 'General, File')
 	 */
 	public NewMenuWizard(String shellText, String[] wizardPath) {
-		super(shellText,wizardPath);
+		super(shellText, wizardPath);
 	}
-	
+
 	@Override
 	protected Openable getOpenAction() {
 		return new SelectionWizardOpenable(new NewWizard(), wizardPath, matcher);
 	}
+
+	/**
+	 * Click the finish button in wizard dialog.
+	 * 
+	 * @param openPerspective open perspective related to the new project
+	 */
+	public void finish(boolean openPerspective) {
+		finish(TimePeriod.LONG, openPerspective);
+	}
+
+	/**
+	 * Click the finish button in wizard dialog.
+	 * 
+	 * @param timePeriod timeout for running job after clicking on Finish button 
+	 * @param openPerspective open perspective related to the new project
+	 */
+	public void finish(TimePeriod timePeriod, boolean openPerspective) {
+		String openAssociatedPerspective = RedDeerProperties.OPEN_ASSOCIATED_PERSPECTIVE.getValue();
+		if (!openAssociatedPerspective.equals("promt")) {
+			throw new RedDeerException("Property " + RedDeerProperties.OPEN_ASSOCIATED_PERSPECTIVE + " is set to '"
+					+ openAssociatedPerspective + "' but it should be set to 'promt'");
+		}
+
+		checkShell();
+		log.info("Finish wizard");
+
+		new FinishButton(this).click();
+
+		try {
+			new WaitWhile(new JobIsRunning(), timePeriod);
+		} catch (NoClassDefFoundError e) {
+			// do nothing, reddeer.workbench plugin is not available
+		}
+		new WaitUntil(new ShellIsAvailable("Open Associated Perspective?"));
```

Comment:
@jkopriva Please, add the check after finish button is pressed, we will get shell opened and preventing other tests to fail, probably.